src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,141 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double GROWTH_FACTOR = 1.5;
+    private transient Object[] elements;
```

Comment:
Why this is transient ?

Reply:
I saw that this keyword is used in the original Arrailist, so I thought it would be appropriate to use it here. As far as I understand, the data of this field will not be saved when trying to transfer data over the network. I'm not sure it's necessary